### DISCUSSION TITLE
Setting multi-relation fields using FormData

### DISCUSSION THREAD

**Discussion by FlixMa** [pocketbase/pocketbase#656] [Q&A] [answered]
When setting a multi-relation field from the js-sdk, pocketbase refuses the creation/update due to invalid formatting when sent through `FormData` instead of the `json` variant.

I fill the `FormData`-value like this: `payload.append('steps', stepRecords.map((record) => { return record.id; }))`

Node-Client Log:
```
FormData {
  [Symbol(state)]: [
    { name: 'image', value: [File] },
    {
      name: 'steps',
      value: '4vobh1tncyzjyso,qsq22semskgdazd,ely4f8336rj7yru,x64rlorhywe5xtz,7xul44tjuynf6gq,t8g4zxyb08o08cw'
    }
  ]
}

ClientResponseError 400: Something went wrong while processing your request. 
{
  url: 'http://127.0.0.1:8090/api/collections/recipes/records/dcrxmfzrgrd7mpy',
  status: 400,
  data: {
    code: 400,
    message: 'Something went wrong while processing your request.',
    data: { steps: [Object] }
  },
  isAbort: false,
  originalError: null
}
```

Pocketbase:
```
[0.00ms] SELECT count(*) FROM `steps` WHERE `id`='4vobh1tncyzjyso,qsq22semskgdazd,ely4f8336rj7yru,x64rlorhywe5xtz,7xul44tjuynf6gq,t8g4zxyb08o08cw'
2022/09/27 15:07:25 steps: Failed to fetch all relation records with the provided ids;
```

When looking up the ids individually using the admin panel I do find them.
To me, this seems as if the serialisation of `FormData` messes with the interpretation on the side of Pocketbase.
Can you confirm -- or is this an error on my side?

**Comment by ganigeorgiev** [comment, marked as answer]
You can't serialize a plain array as FormData value. Instead you have to `append` each id separately, something like:

```js
for (const record of stepRecords) {
    payload.append("steps", record.id)
}
```

_I'll move the issue to Q&A discussion._

**Reply by FlixMa** [reply]
Ok, never mind. Thanks for the heads up!

**Comment by jimafisk** [comment]
Just wanted to share a workaround that worked for me.

I was trying to save an array of IDs to a hidden input using FormData and it was throwing: `Failed to fetch all relation records with the provided ids`

Using a multi-select field allowed me to save an array of IDs to a relation field using `FormData`, I just needed to make sure the IDs were available as option values:

```svelte
<script>
  import { onMount } from 'svelte';

  let allTags = [];
  onMount(async() => {
    allTags = await pb.collection('tags').getFullList();
  });

  let my_array_of_ids = ["bh7rU4y90vE72cS", "0xGrK3uW1vZ6nC1"];
</script>

<select multiple hidden name="my_relation_field" value={my_array_of_ids}>
  {#each allTags as tag}
    <option value="{tag?.id}"></option>
  {/each}
</select>
```
(_Example uses Svelte_)
